extensions/attributegen/config.proto

```diff
@@ -0,0 +1,113 @@
+/* Copyright 2020 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package istio.attributegen;
+
+// Configuration to generate attributes based on existing attributes.
+message PluginConfig {
+  // next id: 2
+  // Multiple independent attribute generation configurations.
+  repeated AttributeGeneration attributes = 1;
+}
+
+// AttributeGeneration uses [well known attributes]
+// (https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/rbac_filter#condition)
+// as inputs and produces a new attribute.
+// The new attribute can be used by a downstream plugin.
+//
+// Example: AttributeGen may produce an attribute called `istio.operationId` which is analogous to
+// the `operationId` in the OpenAPI Spec.
+// The Stats plugin will run after this plugin and use `istio.operationId` to populate a dimension.
+//
+// If multiple AttributeGenerations produce the same attribute, the result of the last generation
+// will be visible to downstream filters.
+message AttributeGeneration {
+  // Phase denotes plugin lifecycle phase when the new attribute is generated.
+  // Default: After the response is sent back to the client.
+  Phase phase = 1;
+
+  // The following attribute is populated on a successful match.
+  // Example: `istio.operationId`
+  // `istio.` attribute namespace is reserved by Istio.
+  //
+  // AttributeGeneration may fail to evaluate when an attribute is not available.
+  // Example: `response.code` may not be available when a request ends abruptly.
+  // When evaluation fails the plugin will set an error_attribute to the error string.
+  // error_attribute = ${output_attribute}_error
```

Comment:
interesting. if i have multiple generators that each produce the same `output_attribute` and only one fails, do i see both an error attribute and a usable attribute ?